internal/utils/generate_scripts.go

```diff
@@ -0,0 +1,188 @@
+package utils
+
+import (
+	"os"
+	"text/template"
+
+	"github.com/NethermindEth/1Click/configs"
+	"github.com/NethermindEth/1Click/internal/templates"
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+GenerateDockerComposeScripts :
+This function is responsible for generating docker-compose scripts for execution, consensus and
+validator clients.
+
+params :-
+a. executionClient string
+Execution client whose script is to be generated
+b. consensusClient string
+Execution client whose script is to be generated
+c. validatorClient string
+Execution client whose script is to be generated
+d. path string
+Path of generated scripts
+
+returns :-
+None
+*/
+func GenerateDockerComposeScripts(executionClient, consensusClient, validatorClient, path string) {
+	_, baseTmp := buildTemplate("docker-compose_base", "base")
+	_, executionTmp := buildTemplate(executionClient, "execution")
+	_, consensusTmp := buildTemplate(consensusClient, "consensus")
+	_, validatorTmp := buildTemplate(validatorClient, "validator")
+
+	tmp := template.Must(template.New("docker-compose").Parse(baseTmp))
+	template.Must(tmp.Parse(executionTmp))
+	template.Must(tmp.Parse(consensusTmp))
+	template.Must(tmp.Parse(validatorTmp))
+
+	// Print docker-compose file
+	log.Infof(configs.PrintingFile, "docker-compose.yml")
+	tmp.Execute(os.Stdout, nil)
+
+	err := writeTemplateToFile(tmp, path+"/docker-compose.yml", nil, false)
+	if err != nil {
+		log.Fatalf(configs.GeneratingScriptsError, executionClient, consensusClient, validatorClient)
```

Comment:
return errors to the calling function and handle them gracefully instead of panic

Reply:
ok